### DISCUSSION TITLE
A Community Discord

### DISCUSSION THREAD

**Discussion by EverettHanke** [eclipse-pde/eclipse.pde#1512] [Ideas]
CodeDay student here, so my opinion might not hold much weight, but I think the Eclipse community could really benefit from having a community Discord server. Not necessarily for solving issues directly, but for fostering a sense of community in general. Imagine being able to sit in an empty voice chat and occasionally seeing someone else pop in getting to talk to a complete stranger with similar interests. It could be an excellent way to create friendships while also becoming an invaluable resource for newcomers to any open-source project. They could ask quick questions like, "_How do I merge my new changes into my current PR without accidentally breaking the active PR?_" or "_My development environment isn’t working, did I miss a step?_"

I know this isn’t just a personal issue. Over the past three months, my cohort was assigned to contribute to either Eclipse or Open Energy Dashboard. Half the cohort (the Eclipse side) struggled with insignificant issues that could have been resolved with a stronger community presence—simple problems like setting up the development environment or reconfiguring it to work on Mac. We even had an entire team abandon Eclipse in favor of Open Energy Dashboard due to these challenges. Those of us who stuck with Eclipse barely managed to get anything done, if at all. Meanwhile, the Open Energy Dashboard students were completing issues nearly every other week, largely thanks to their community Discord, where they could easily ask for guidance and clarification.

There was a real sense of community on the Open Energy Dashboard side—a shared purpose. Their students are heading into winter break excited to keep contributing, while only one other student and I want to keep working with Eclipse outside of school. A running joke at my college is that "_Eclipse is named Eclipse because the community is left in the dark_," and while I don’t want to agree with that, it’s hard to argue otherwise.

I’m not trying to compare communities just to say how much better the other side is. I understand every community has its own approach. But not having a Discord server seems like a serious missed opportunity. This wasn’t a bad experience because of technical issues—it was the lack of community. Unlike my peers, I don’t want to quit or give up on the community. Maybe I’m stubborn, but I genuinely believe that for the benefit of newcomers and the community as a whole, we need something more than just GitHub discussions.

Anyways, my name is Everett. Thanks for listening to my TED Talk. I hope to hear something back—anything, really.

**Comment by HannesWell** [comment]
Hi Everett,
thank you for your feedback and your persistent interest in Eclipse. I'm surprised about the situation you described. The feedback we received from CodeDay was in general positive and even though technical problems with the setup of the the development environments (and similar) were reported, a lack of responsiveness and support wasn't a complain. Still I'm aware that two/three PRs didn't receive the attention they need and are awaiting feedback and I'm trying my best to provide it soon.

To your questions about a direct chat:
In general the page https://github.com/eclipse-ide should provide you all information you need as a newcomer, it is intended as landing-page for new contributors. But please note that PDE is only one part of the Eclipse-IDE. And specifically, we have a Matrix chat hosted by the Eclipse Foundation as described in:
- https://github.com/eclipse-ide/.github/blob/main/help.md#-questions-about-contributions
There is even a room, dedicated for newcomers.

Furthermore, if you want to give feedback about your contributing experience you can also use
- https://github.com/eclipse-ide/contributing

The entire https://github.com/eclipse-ide page was under construction, while you students from CodeDay started and you unfortunately missed it in the beginning.

> It could be an excellent way to create friendships while also becoming an invaluable resource for newcomers to any open-source project. They could ask quick questions like, "How do I merge my new changes into my current PR without accidentally breaking the active PR?" or "My development environment isn’t working, did I miss a step?"

While I agree that it's crucial for an open-source project like Eclipse to attract contributors, which of course implies on-boarding newcomers, we hope that these newcomers are already proficient or have dedicated 'external' mentors to assist them like in case of CodeDay.
So we were hoping that you students from CodeDays would get such fundamental questions answered by your CodeDay mentor. We consider our main priority in answering the technical questions and reviewing PR contents respectively making newcomers familiar with the specialties of Eclipse and it's development process. But we cannot help much with basic git or github tasks or teach for example basic Java, simplify because we don't have the time for that. Everyone is busy and resources are limited and helping at that level takes much more time. It's nice to do it, please don't get me wrong, but we just cannot offer that.

But your specific problem regarding unintended closing PRs probably exists because you use the `master` branch of your fork as branch for a PR and as soon as you sync your fork, the PR-branch contains zero new commits and gets closed.

**Comment by tfroment** [comment]
Hi @EverettHanke 
Thank you so much for sharing your perspective: feedback is always a gift, whether it’s positive or negative, perhaps even more so when it’s negative, as it provides an opportunity to learn and improve...

I completely agree that fostering a stronger sense of community is critical, and I understand the challenges you and your cohort faced. From my point of view, one of the unique difficulties in the Eclipse IDE ecosystem stems from the sheer number of projects and contributors involved. With over 80 projects and 118 million lines of code in what we generally call the Eclipse IDE, it’s undeniably complex.

It’s because of this complexity that we’ve put mentors like @HannesWell and @opcoach in place to help support the community. We want contributors to feel they have the guidance they need to navigate these challenges.

Regarding the chat tool, while we don’t have a Discord server, we do have a Matrix chat (https://chat.eclipse.org/#/room/#eclipse-ide:matrix.eclipse.org) where the community is active and responsive. Have you had a chance to check it out? 
Your feedback is invaluable because it reminds us of the importance of continuously evolving. As Hannes mentioned, the new [GitHub organization](https://github.com/eclipse-ide) was still under construction when he joined, and it’s still improving, thanks to input like yours. It’s these transparent and honest insights that help guide us in the right direction.

I hope we can convince you that things are just beginning to evolve for the better, and I truly hope you’ll stay engaged with the community. Contributors like you, who share their experiences and ideas openly, are a tremendous asset, and your dedication inspires us to create a stronger, more connected Eclipse platform ecosystem: we’re currently at the beginning of a major technical renovation of this ecosystem, called [Initiative31](https://github.com/swt-initiative31), and I’d really encourage you to take a look at this [video](https://www.youtube.com/watch?v=4b1_gM4SmOA&t=428s).  I am convinced that contributing to this new generation of an ecosystem that has achieved over 300 million downloads is a significant and motivating opportunity: I mean... while I have nothing against “Open Energy Dashboard”, but it has nothing to compare in terms of impact with the Eclipse ecosystem - it’s worth noting that some of the most critical systems in the world, like the programs that take us to Mars (just to mention one example), still rely on these Eclipse Open Source projects - I personnally thinkg that’s an inspiring thought, beyond the initial complexity!

Thank you again for your thoughtful feedback, and I look forward to continuing this conversation, feel free to reach out to me...

**Reply by Gamebuster19901** [reply]
...And why exactly do I need to provide my real name and employment status (and the name of the place where I work?!) to ask questions in a chat room? 

I don't even work in the software industry. Providing information like that for no reason is how someone gets doxxed.

**Comment by mickaelistria** [comment]
While I'm not much following the IM trends and activity, I think Discord may be a better (as in more popular and more accessible) channel than chat.eclipse.org which require people to already be aware of its existence. So I find the proposal very valid, and I would be curious know, out of all the people who currently use chat.eclipse.org for Platform/PDE development, how many would be OK to use discord instead?
If Discord is also preferred by current maintainers who use chat.eclipse.org, then it would be pragmatically beneficial to use it instead of chat.eclipse.org.

**Reply by tfroment** [reply]
You raise a valid point about Discord potentially being a more popular and accessible option compared to chat.eclipse.org. That said, no system is perfect, and tool selection often comes with trade-offs. For example, in another Working Group (and in the platform team as well ;-)), we’ve seen resistance to moving away from Slack (even with its 30-day message history limitation), simply because people are already familiar with it and hesitant to juggle multiple tools.
In the corporate world, Discord often faces challenges: it’s not particularly popular and, in many cases, is outright blocked. On the other hand, while Matrix chat isn’t flawless, it has the benefit of being integrated with Eclipse’s SSO... So... theoritically, anyone who starts working on an Eclipse project already has access without the need to set up or manage additional accounts...

I’m not saying that Discord or any other tool isn’t a good idea, but my perception is that regardless of the tool chosen, its success ultimately depends on active participation from the community. Without engaged contributors driving activity, even the most popular or technically advanced tool won’t guarantee better engagement or outcomes.

As for gauging interest in a potential switch, your suggestion to understand how many current Matrix users for Platform/PDE development would be open to using Discord instead is a great idea?

**Reply by mickaelistria** [reply]
I just learned that Discord is not available in China. That's IMO a major issue.
